command/update/command.go

```diff
@@ -269,8 +201,48 @@ func (c *Command) execute() error {
 			return microerror.MaskAny(err)
 		}
 
-		c.logger.Log("debug", fmt.Sprintf("updated endpoint for service '%s'", f.Kubernetes.Cluster.Service))
+		c.logger.Log("debug", fmt.Sprintf("added IP to endpoint of service '%s'", f.Kubernetes.Cluster.Service))
 	}
 
+	// Listen to OS signals issued by the Kubernetes scheduler.
+	listener := make(chan os.Signal, 2)
+	signal.Notify(listener, syscall.SIGTERM, syscall.SIGKILL)
+
+	c.logger.Log("debug", "waiting for termination signal")
+
+	s := <-listener
+
+	c.logger.Log("debug", fmt.Sprintf("received termination signal: %#v (%s)\n", s, s))
```

Comment:
The execution of the endpoint updater is now directly done without shell script wrapping. See https://github.com/giantswarm/kvm-operator/pull/203#discussion_r140665654. Now we listen to termination signals and remove the registered IP when we git asked to shutdown the process.